### PR TITLE
Overhaul external / binplacing / reference from runtime

### PR DESCRIPTION
**Pull request by ericstj**
Best to review commit-by-commit.

First change I made was to switch external to use binplacing logic.  Let let us have better control over when to copy and where and allowed copying to multiple locations easily.  As part of this I binplace both the primary output (if it exist) as well as all copy-local items.

Second change was to combine coreclr and NETNative into a common runtime project  (thanks @joperezr).  This was to enable the following change.

Third I fixed ReferenceFromRuntime to use a project reference to the runtime project for src projects, and block its use from tests.  This enables us to build multiple configurations at a time that may need different runtime libs without stomping on each other in the runtime path.  It also fixes cases where folks might accidentally use ReferenceFromRuntime instead of a project reference.

Finally I noticed that the first change caused a bunch of clashing copies to occur.  To fix this (which should be general goodness) I disabled copy local for project references for anything that isn't a test project.

/cc @weshaggard @tarekgh

### CODE REVIEW

File: external/coreclr/coreclr.depproj [hunk 1]
```diff
@@ -2,8 +2,6 @@
 <Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" DefaultTargets="Build">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
-    <OutputPath>$(RuntimePath)</OutputPath>
-     <!-- WORKAROUND: Force external packages to be restored for x64 until arm packages are fully broughtup-->
```
Comment (mellinoe) [inline external/coreclr/coreclr.depproj]: Intentionally removed the comment?

Comment (mellinoe) [inline external/coreclr/coreclr.depproj]: I see the file was consolidated, but maybe we still want the comment in the new location (runtime.depproj)

**Comment by mellinoe** [comment]
This LGTM. The previous functionality was indeed dangerous if you used it to reference anything violating the project build ordering, and there was nothing stopping that from happening.

File: external/coreclr/coreclr.depproj (hunk 1 above)
Comment (ericstj) [inline external/coreclr/coreclr.depproj]: I can add that back, removing the comment was accidental.

Comment (ericstj) [inline external/coreclr/coreclr.depproj]: I doctored up the commits to bring that comment back 👍

**Pull request merged**